model-server/src/test/kotlin/org/modelix/model/server/ReplicatedModelTest.kt

```diff
@@ -0,0 +1,154 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.server
+
+import io.ktor.server.testing.ApplicationTestBuilder
+import io.ktor.server.testing.testApplication
+import kotlinx.coroutines.delay
+import org.modelix.authorization.installAuthentication
+import org.modelix.model.IVersion
+import org.modelix.model.api.ChildLinkFromName
+import org.modelix.model.api.ConceptReference
+import org.modelix.model.api.IBranch
+import org.modelix.model.api.ITree
+import org.modelix.model.api.PBranch
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client2.ModelClientV2
+import org.modelix.model.client2.ReplicatedModel
+import org.modelix.model.lazy.BranchReference
+import org.modelix.model.lazy.CLTree
+import org.modelix.model.lazy.CLVersion
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.operations.OTBranch
+import org.modelix.model.server.handlers.ModelReplicationServer
+import org.modelix.model.server.store.InMemoryStoreClient
+import java.util.UUID
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class ReplicatedModelTest {
+
+    @Test
+    fun startsWithLatestVersion() = runTest {
+        val client = createModelClient()
+        val repositoryId = RepositoryId(UUID.randomUUID().toString())
+
+        // Step 1: prepare repository with two versions beside the initial version
+        // Step 1.1: create an empty repository
+        val initialVersion = client.initRepository(repositoryId)
+
+        // Step 1.2: add a new child node to get a new version
+        val defaultBranchReference = repositoryId.getBranchReference()
+        val oneHelloChildVersion = addHelloChild(initialVersion, client, defaultBranchReference)
+
+        // Step 1.3: add a new child node to get a newer version
+        val twoHelloChildrenVersion = addHelloChild(oneHelloChildVersion, client, defaultBranchReference)
+
+        // Step 2: in a new client, fetch the latest repository data
+        val newClient = createModelClient()
+        // we do not provide an initial version, so we expect to fetch the latest one (twoHelloChildrenVersion)
+        val replicatedModel = ReplicatedModel(newClient, defaultBranchReference)
+        val branch = replicatedModel.start { branch ->
+            // Step 3: check, here we must have the twoHelloChildrenVersion (2 "hello" children)
+            val twoHelloChildren = getHelloChildrenOfRootNode(branch)
+            assertEquals(2, twoHelloChildren.size)
+        }
+
+        // Step 4: wait a bit so replicated model can fetch the new versions from the server (in case there would be any -- there is none by the way)
+        delay(500)
```

Comment:
This looks like it has the potential to result in a flaky unit test if we need waiting for synchronizing. Moreover, from the `runTest` docs:

> On JVM and Native, this function behaves similarly to runBlocking, with the difference that the code that it runs will skip delays. This allows to use [delay](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/index.html) in tests without causing them to take more time than necessary. On JS, this function creates a Promise that executes the test body with the delay-skipping behavior.

So do we really need this here?

Reply:
removed in [481f83d](https://github.com/modelix/modelix.core/pull/725/commits/481f83da1fef69c3b946773d198703fcae70ce8e)